smartsim/builders/ensemble.py

```diff
@@ -374,23 +357,47 @@ def _create_applications(self) -> tuple[Application, ...]:
         :return: A tuple of Application instances
         """
         permutation_strategy = strategies.resolve(self.permutation_strategy)
-
-        combinations = permutation_strategy(
-            self.file_parameters, self.exe_arg_parameters, self.max_permutations
+        file_set_list: list[list[FileSet]] = [
+            self._permutate_file_parameters(config_file, permutation_strategy)
+            for config_file in self.files.configure_operations
+        ]
+        file_set_tuple: list[tuple[FileSet, ...]] = self._cartesian_values(
+            file_set_list
         )
-        combinations = combinations if combinations else [ParamSet({}, {})]
         permutations_ = itertools.chain.from_iterable(
-            itertools.repeat(permutation, self.replicas) for permutation in combinations
+            itertools.repeat(permutation, self.replicas)
+            for permutation in file_set_tuple
         )
-        return tuple(
-            Application(
+        app_list = []
+        for i, item in enumerate(permutations_, start=1):
+            app = Application(
                 name=f"{self.name}-{i}",
                 exe=self.exe,
                 exe_args=self.exe_args,
-                file_parameters=permutation.params,
             )
-            for i, permutation in enumerate(permutations_)
-        )
+            self._attach_files(app, item)
+            app_list.append(app)
+        return tuple(app_list)
+
+    def _attach_files(
+        self, app: Application, file_set_tuple: tuple[FileSet, ...]
+    ) -> None:
+        """Attach files to an Application.
```

Comment:
nitpick: again, we're attaching the operations, not the files themselves

Reply:
all nitpicks welcome always!